File: elixir/bob/lib/bob.ex
defmodule Bob do
  def hey(input) do
    cond do
      nothing?(input) -> "Fine. Be that way!"
      question?(input) && shouting?(input) -> "Calm down, I know what I'm doing!"
      shouting?(input) ->  "Whoa, chill out!"
      question?(input) ->  "Sure."
      true -> "Whatever."
    end
  end

  def nothing?(input) do
    String.trim(input) == ""
  end

  def question?(input) do
    String.last(input) == "?"
  end

  def shouting?(input) do
    input == String.upcase(input) && String.match?(input, ~r/[\p{L}]/)
  end 
end

File: typescript/bob/bob.ts
const isQuestion = (str: string): boolean => str.endsWith("?");
const isShouting = (str: string): boolean =>
  /[A-Z]+/.test(str) && str === str.toUpperCase();

export const hey = (message: string): string => {
  let response = "Whatever.";
  const trimmedMsg = message.trim();

  if (trimmedMsg === "") response = "Fine. Be that way!";
  else if (isQuestion(trimmedMsg) && isShouting(trimmedMsg))
    response = "Calm down, I know what I'm doing!";
  else if (isShouting(trimmedMsg)) response = "Whoa, chill out!";
  else if (isQuestion(trimmedMsg)) response = "Sure.";

  return response;
};
